### PR TITLE
Fix order of bft execution - Closes #7060

### LINKED ISSUE #7060: BFT module fails after initial round
### Expected behavior
- Update `afterTransactionsExecute` hook to be `beforeTransactionsExecute` so that other module can expect the BFT votes are updated when calling `setBFTParameters`
- During updating the BFT votes it should not fail if the block generator is not in the BFT parameter https://github.com/LiskHQ/lisk-sdk/blob/development/framework/src/modules/bft/bft_votes.ts#L91-L95 and https://github.com/LiskHQ/lisk-sdk/blob/development/framework/src/modules/bft/bft_votes.ts#L126-L131 because it could be validator without BFT vote weight (ex: standby delegate)

### Actual behavior
- When DPoS calls setBFTParameters, it expects bftVotes to be executed for the block.  However, with the current module order, BFT module hook will be called later since it is system module.
- If a block is generated by a generator who does not have bft weight, it will fail

### Steps to reproduce
Run a node more than initial rounds

### Which version(s) does this affect? (Environment, OS, etc...)
development

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #7060 

### How was it solved?

- Update hook to be executed in `beforeTransactionsExecute`
- replace BFTValidator not exist error with continue. It is possible that generator address is not bft participant but valid generator of a block.

### How was it tested?
- Updated unit tests
- Run application more than 309 blocks from genesis

### CODE REVIEW

File: framework/src/modules/bft/bft_votes.ts [hunk 1]
```diff
@@ -123,11 +119,7 @@ export const updatePrevotesPrecommits = async (
 			v.address.equals(blockBFTInfo.generatorAddress),
 		);
 		if (!bftValidator) {
-			throw new Error(
-				`Invalid state. Validator ${blockBFTInfo.generatorAddress.toString(
-					'hex',
-				)} must be in the BFT parameters at height ${blockBFTInfo.height}`,
-			);
+			continue;
```
[the remark this answers is not in the document]
Comment (janhack) [reply framework/src/modules/bft/bft_votes.ts:122]: Same as above.

File: framework/src/modules/bft/bft_votes.ts [hunk 2]
```diff
@@ -88,11 +88,7 @@ export const updatePrevotesPrecommits = async (
 				v.address.equals(blockBFTInfo.generatorAddress),
 			);
 			if (!bftValidator) {
-				throw new Error(
-					`Invalid state. Validator ${blockBFTInfo.generatorAddress.toString(
-						'hex',
-					)} must be in the BFT parameters at height ${blockBFTInfo.height}`,
-				);
+				continue;
```
[the remark this answers is not in the document]
Comment (janhack) [reply framework/src/modules/bft/bft_votes.ts:91]: I updated my comment above.

**Review by janhack — CHANGES_REQUESTED** [review]
As commented, I think there is still a minor issues with respect to the computation of prevotes and precommits.

**Review by mitsuaki-u — APPROVED** [review]

**Review by janhack — COMMENTED** [review]
Note that we have the line
```
if (activeValidatorIndex > -1) {
```
This should always be `true` as otherwise `validatorInfo` before would not be defined. So the code is correct, but this additional if condition would not be necessary.

**Review by janhack — APPROVED** [review]

**Pull request merged**